firebase/firebase-unity-sdk
# Mila/MultiDB

Add multiDB support.

Ported from cpp SDK: https://github.com/firebase/firebase-cpp-sdk/pull/1321

TODO:
 - [x] Add 2 new Firestore::GetInstance public interface to accept dynamic database name
- [x]  Update the firestore cache to use new structure `FirestoreInstanceCacheKey` as key instead of `FirebaseApp` to identify databases in the same app while creating/deleting firestore instances.
 - [x] Add test cases for multiDB support
- New GetInstance method call verification
- Instances' equality check
- Can create & terminate multiple Firestore instances (commented out)
- Can create and fetch documents from different Firestore instances (commented out)
- Can listen to different Firestore instances in Parallel(commented out)
- Can run transactions in different Firestore instances in Parallel(commented out)
- Can terminate Firestore instances independently, or by app disposal, and restart (commented out)

How to test locally:
1. fire up the local firestore emulator
2. Uncomments the following line of code in `firestore/testapp/Assets/Firebase/Sample/Firestore/UIHandlerAutomated.cs` to set the `USE_FIRESTORE_EMULATOR` environment variable: 
 `Environment.SetEnvironmentVariable("USE_FIRESTORE_EMULATOR", "true");`

3. The port number is default to “8080”. If your Firestore emulator is running on a different port number, add the following line of code to modify `FIRESTORE_EMULATOR_PORT`
` Environment.SetEnvironmentVariable("FIRESTORE_EMULATOR_PORT", <port_number>);`

4. Run the `UIHandelerAutomated` tests again. This time all tests should run against the local Firestore emulator.
To switch back to running tests on the production backend, set the `USE_FIRESTORE_EMULATOR` to null.
